### PR TITLE
Merge in 1.4.2 fixes - verified 

### PR DESCRIPTION
**Pull request by JacobGros**
* Verified merge fixed release v1.4.2 (#479)

* Add permissions for SC removal. (#431)

* Bump_up_csm-operator_version_to_1.4.1 (#434)

* Patch version support (#433)

* Adding support of v2.9.1 for Pmax, Pscale, PFlex and unity

* Updating module config version for Pmax, Pscale, PFlex and unity

* Update storage_csm_powerflex.yaml

* version changed to 2.9.1

* version changed to 2.9.1

* Enabling github actions for all branch

* updated version to 2.9.1 for unity, powerflex, powerscale and powermax

* Adding the support of PStore v2.9.1

* Adding the support of PFlex v2.9.1

* Version bumpup replication 1.7.1

* Bump up version for authorization, resiliency

* update csi-metadata-retriever version to 1.6.1

* update reverseproxy version to 2.8.1

---------

* Change namespace name to match documentation. (#437)

* go-yaml version update (#438)

* go-yaml version update

* update go-yaml package to latest

* remove extra verbs

* remove duplicate verbs and add correct test files for resiliency UT

---------

* [bug-1120]: Update Authorization sidecar to use insecure flag (#442)

* fix auth cert

* remove constant

* remove duplicate verbs

* reverting #426 changes (#444)

* Update final manifests for posting of Operator 1.4.1 (#445)

* KRV-20573/csm status (#446)

* add module status

* Adding CSM Module status

* Addressing the comments for PR

* Making code reusable

* Fix formatting issue

* Resolving gosec issue

* Final manifest update and apex client (#447)

* Add csm watch permissions. (#448)

* Merged CSM Module Status

* Fix App-Mob node error (#457)

* Bump up CSM Operator version to 1.4.2 (#458)

* update csm operator version to 1.4.2

* bump-up-op-version-1.4.2

* updated version in operator.yaml

* adding missing arg to fix e2e

* update authorization e2e (#459)

* skip cert validation in e2e (#460)

* Fixing FSGroupPolicy issue (#463)

* Fixing FSGroupPolicy issue

* Add unit test

---------

* Fix for controller race, daemon status, and obs namespace  (#464)

* test controller state calc

* add debug logs

* check for driver

* add daemonnset update

* adds ns fix for when pod ns dis not csm ns

* add ns check for daemon and deployment as well

* fix typo

* fix typo

* explicitly return nil

* add missing replace statements

---------

* Auth proxy status fix (#465)

* actions

* working on auth proxy status check

* finished status check

* fix 1

* fix or

* code compiles

* keep auth proxy out of daemonset check

* trying to figure out deployment name

* formatting

* fix auth name in if condition

* no daemonset

* fix

* added log statements

* added another log statement

* fix log statement

* logs logs logs

* compile

* more logs

* add logs and try updating status from handlesuccess

* remove UpdateStatus call for now

* adding csm label to auth

* remove extraneous string replacement

* removing commented code

* Update actions.yml

* fix merge issue

* fix fmt

* fix pscale + obs issue  (#467)

* fix some issues

* fiix label issue, add debug logs

* add more debug

* more  debugg lines

* fix pscale issue

* remmove debug logs

* gofmt

* set available to 0 en deployment is not checked

* fix if check

---------

* update final manifests (#468)

* fix vetting issues (#477)

---------

* PR feedback

---------

# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1146 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran E2e
- [x] Ran Unit

### CODE REVIEW

File: operatorconfig/driverconfig/powerflex/v2.8.0/node.yaml [hunk 1]
```diff
@@ -109,11 +109,6 @@ spec:
               value: <X_CSI_RENAME_SDC_PREFIX>
             - name: X_CSI_MAX_VOLUMES_PER_NODE
               value: <X_CSI_MAX_VOLUMES_PER_NODE>
-            - name: X_CSI_POWERFLEX_KUBE_NODE_NAME
-              valueFrom:
-                fieldRef:
-                  apiVersion: v1
-                  fieldPath: spec.nodeName
```
Comment (panigs7) [inline operatorconfig/driverconfig/powerflex/v2.8.0/node.yaml:116]: These changes should be reverted. As it was a fix raised in [PR](https://github.com/dell/csm-operator/pull/426/files). Please follow the same for other node.yaml files for powerflex (v2.7.0, v2.9.0 and testfiles)

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -3520,7 +3520,7 @@ spec:
     name: Dell Technologies
     url: https://github.com/dell/csm-operator
   relatedImages:
-  - image: docker.io/dellemc/dell-csm-operator:v1.4.1
+  - image: docker.io/dellemc/dell-csm-operator:v1.4.2
```
Comment (panigs7) [inline bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:3523]: Version should be 1.5.0

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -3575,5 +3575,5 @@ spec:
   - image: docker.io/dellemc/connectivity-cert-persister-k8s:0.7.0
     name: cert-persister
   skips:
-    - dell-csm-operator.v1.4.0
-  version: 1.4.1
+  - dell-csm-operator.v1.4.0
+  version: 1.4.2
```
Comment (panigs7) [inline bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:3579]: Version 1.5.0 instead of 1.4.2

**Review by jooseppi-luna — DISMISSED** [review]
Just as an update, ran all PFlex E2E tests and only failed on one of the last app mobility scenarios because a docker image pull of an artifactory image was acting up (not operator related at all). Did find a weird edge case with PowerScale where the controller failed due to the csi-metadata-retriver v1.7.0 image not existing yet, but the csm status was marked as successful. I don't think this is a regression being introduced in this PR, so I will approve the PR.

```
[root@master-1-Zaglt7mQUY8Wg observability]# kubectl get pods -n isilon
NAME                                 READY   STATUS    RESTARTS       AGE
isilon-controller-67c5c64b86-bptxg   1/6     Error     12 (14s ago)   2m52s
isilon-node-2h9bt                    2/2     Running   0              2m52s
isilon-node-mtdmd                    2/2     Running   0              2m52s
[root@master-1-Zaglt7mQUY8Wg observability]# kubectl get csm -A
NAMESPACE   NAME     CREATIONTIME   CSIDRIVERTYPE   CONFIGVERSION   STATE
isilon      isilon   2m55s          isilon          v2.10.0         Succeeded
[root@master-1-Zaglt7mQUY8Wg observability]#
```

File: operatorconfig/driverconfig/powerflex/v2.8.0/node.yaml (hunk 1 above)
Comment (jooseppi-luna) [reply operatorconfig/driverconfig/powerflex/v2.8.0/node.yaml:116]: @panigs7 will do I will revert these now

File: pkg/utils/status.go [hunk 4]
```diff
@@ -881,3 +965,104 @@ func observabilityStatusCheck(ctx context.Context, instance *csmv1.ContainerStor
...
+	}
+
+	checkFn := func(deployment *appsv1.Deployment) bool {
+		return deployment.Status.ReadyReplicas == *deployment.Spec.Replicas
+	}
+
+	for _, deployment := range deploymentList.Items {
+		deployment := deployment
+		switch deployment.Name {
+		case "authorization-ingress-nginx-controller":
```
Comment (atye) [inline pkg/utils/status.go:1004]: This needs to be dynamic for the namespace, unless we are reverting that change.

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (jooseppi-luna) [reply bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:3523]: @panigs7 why is it 1.4.1 in main? When/how does this get updated? There's a lot of ourdated driver versions and stuff in there as well. I don't think this is an issue for this PR to solve. We will revert to 1.4.1 as it currently is in main.

File: pkg/utils/status.go (hunk 4 above)
Comment (jooseppi-luna) [reply pkg/utils/status.go:1004]: that change is coming after this PR goes in @atye

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (jooseppi-luna) [reply bundle/manifests/dell-csm-operator.clusterserviceversion.yaml]: @panigs7 undid these changes, please re-review

**Review by jooseppi-luna — APPROVED** [review]

**Review by atye — APPROVED** [review]

File: pkg/utils/status.go [hunk 5]
```diff
@@ -767,6 +844,14 @@ func observabilityStatusCheck(ctx context.Context, instance *csmv1.ContainerStor
 	metricsRunning := false
 	topologyRunning := false
 
+	driverName := instance.Spec.Driver.CSIDriverType
+
+	// TODO: PowerScale DriverType should be changed from "isilon" to "powerscale"
```
Comment (alikdell) [inline pkg/utils/status.go:849]: Remove TODO

Comment (jooseppi-luna) [reply pkg/utils/status.go:849]: Why? It is a current TODO

Comment (alikdell) [reply pkg/utils/status.go:849]: We aren't suppose to have TODO in checkin code

Comment (jooseppi-luna) [reply pkg/utils/status.go:849]: Hmm ok didn't realize that -- where is the documentation on that, or is it just an assumed standard? How do we document something like this that we are planning on fixing in an upcoming release?
